BeeSwiftTests/GoalViewModelTests.swift

```diff
@@ -0,0 +1,86 @@
+//
+//  GoalViewModelTests.swift
+//  BeeSwiftTests
+//
+//  Created by krugerk on 2024-11-25.
+//
+
+import Testing
+
+@testable import BeeSwift
+@testable import BeeKit
+
+struct GoalViewModelTests {
+    
+    @Test func initialStepperIsMinusOneWhenSubmissionDateIsAfterMidnightAndBeforeDeadline() async throws {
+        let goal = Self.makeGoalWithDeadline(3600 * 3)
+        let viewModel = GoalViewModel(goal: goal)
+        let submissionDate = Calendar.current.date(bySettingHour: 1, minute: 30, second: 0, of: Date())!
+        let actual = viewModel.initialDateStepperValue(date: submissionDate)
+        #expect(actual == -1)
+    }
+    
+    @Test func initialStepperIsZeroWhenSubmissionDateIsBeforeMidnightAndBeforeDeadline() async throws {
+        let goal = Self.makeGoalWithDeadline(0)
+        let viewModel = GoalViewModel(goal: goal)
+        let submissionDate = Calendar.current.date(bySettingHour: 20, minute: 30, second: 0, of: Date())!
+        let actual = viewModel.initialDateStepperValue(date: submissionDate)
+        #expect(actual == 0)
+    }
+
+    @Test func initialStepperIsPlusOneWhenSubmissionDateIsAfterDeadline() async throws {
+        let goal = Self.makeGoalWithDeadline(3600 * -3)
+        let viewModel = GoalViewModel(goal: goal)
+        let submissionDate = Calendar.current.date(bySettingHour: 22, minute: 30, second: 0, of: Date())!
+        let actual = viewModel.initialDateStepperValue(date: submissionDate)
+        #expect(actual == 1)
+    }
+}
+
+
+
+private extension GoalViewModelTests {
+    static func makeGoalWithDeadline(_ deadline: Int) -> Goal {
+        let context = BeeminderPersistentContainer.createMemoryBackedForTests().newBackgroundContext()
```

Comment:
I wonder how CoreData contexts are meant to work in tests

Reply:
In order for the correct value to appear in the view, some place has to know 'submission date aka now' and a goal's deadline. As is, this means knowing everything about a goal and thus about core data.

If the core data part could be more limited - maybe like in the mr for widgets #widgets - where there are conversions from nsmanagedobject goal to a smaller entity, containing just the parts that screen (widget) needs, then test setup and testing pieces should be easier and more focused.